tools/importTool/src/main/scala/AkkaFileReaderWithActor.scala

```diff
@@ -0,0 +1,138 @@
+
+  implicit val mat = ActorMaterializer()
+  val bytesAccumulatorActor = this.system.actorOf(Props(new BytesAccumulatorActor()), name = "offsetActor")
+  context.watch(bytesAccumulatorActor)
+
+  override def receive: Receive = {
+    case ActorInput => {
+      println("Starting flow.....")
+      readAndImportFile(inputUrl, format, cluster)
+    }
+    case Terminated(bytesAccumulatorActor) => {
+      println("BytesAccumulatorActor has been crashed, going to restart the flow")
+      readAndImportFile(inputUrl, format, cluster)
+
+    }
+  }
+
+  def readFileFromServer(inputUrl:String) = {
+    println("Going to send get request to inputUrl=" + inputUrl)
+      val future = bytesAccumulatorActor ? LastOffset
+      future.flatMap(x => {
+        val ranges: String = "bytes=" + x + "-"
+        println("Resending get request for ranges=" + ranges)
+        val request = Get(inputUrl).addHeader(RawHeader.apply("Range", ranges))
+        Http().singleRequest(request)
+      })
+  }
+
+  def readAndImportFile(inputUrl:String, format:String, cluster:String)= {
+    val httpResponse = readFileFromServer(inputUrl)
+    httpResponse.onComplete(
+      {
+        case Success(r) if  r.status.isSuccess() => println("Get file from server successfully")
+        case Success(r) if  r.status.isFailure() => println("Failed to read file, got status code:" + r.status)
+        case Failure(e) => {
+          println("Got a failure while reading the file, ", e.getMessage)
+          system.scheduler.scheduleOnce(7000.milliseconds, self, ActorInput)
+        }})
+    httpResponse.foreach({ response =>
+      println("Get response status=" + response.status)
+      val contentLengthHeader = response.headers.find(_.name == "Content-Range").map(_.value).getOrElse("")
+      val fileSize = contentLengthHeader.split("/")(1).toLong + 1
+      bytesAccumulatorActor ! FileSize(fileSize)
+      response.entity.withSizeLimit(fileSize).dataBytes
+        .via(Framing.delimiter(ByteString(System.lineSeparator()), 100000, true))
+        .map(_.utf8String)
+        .map(getTripleKey)
+        .sliding(2, 1)
+        .splitAfter(pair => (pair.head, pair.last) match {
+        case (TripleWithKey(sub1, triple1), TripleWithKey(sub2, triple2)) => sub1 != sub2
+      })
+        .mapConcat(_.headOption.toList)
+        .fold("" -> List.empty[String]) {
+        case ((_, values), TripleWithKey(subject, triple)) => subject -> (triple :: values)
+      }
+        .map(x => x._2)
+        .mergeSubstreams
+        .grouped(25)
+        .mapAsync(1)(batch=> postWithRetry(batch, format, cluster))
+        .map(x=> bytesAccumulatorActor ! Message(x._2))
+        .runWith(Sink.ignore)
+        .onComplete(
+          {
+            case Success(r) => {
+              println("Process completed successfully")
+              system.terminate()
+            }
+            case Failure(e) => {
+              println("Got a failure during the process, ", e.getMessage)
```

Comment:
you should print the whole stack trace

Reply:
get message prints the whole message whilr get stack trace prints Ljava.lang.StackTraceElement;@6127e450